src/components/TransformWorkflow/FlowEditor/Components/OperationPanel/Forms/DropColumnOpForm.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { OperationNodeData } from '../../Canvas';
+import { useSession } from 'next-auth/react';
+import {
+  Autocomplete,
+  Box,
+  Button,
+  Grid,
+  TextField,
+  Typography,
+} from '@mui/material';
+import { OPERATION_NODE, SRC_MODEL_NODE } from '../../../constant';
+import { DbtSourceModel } from '../../Canvas';
+import { httpGet, httpPost } from '@/helpers/http';
+import { ColumnData } from '../../Nodes/DbtSourceModelNode';
+import { GlobalContext } from '@/contexts/ContextProvider';
+import IconButton from '@mui/material/IconButton';
+import CloseIcon from '@mui/icons-material/Close';
+import InputAdornment from '@mui/material/InputAdornment';
+import {
+  errorToast,
+  successToast,
+} from '@/components/ToastMessage/ToastHelper';
+import { OperationFormProps } from '../../OperationConfigLayout';
+
+const DropColumnOp = ({
+  node,
+  operation,
+  sx,
+  continueOperationChain,
+}: OperationFormProps) => {
+  const { data: session } = useSession();
+  const [srcColumns, setSrcColumns] = useState<string[]>([]);
+  const globalContext = useContext(GlobalContext);
+  const [selectedColumns, setSelectedColumns] = useState<string[]>([]);
+  const nodeData: any =
+    node?.type === SRC_MODEL_NODE
+      ? (node?.data as DbtSourceModel)
+      : node?.type === OPERATION_NODE
+      ? (node?.data as OperationNodeData)
+      : {};
+
+  const fetchAndSetSourceColumns = async () => {
+    if (node?.type === SRC_MODEL_NODE) {
+      try {
+        const data: ColumnData[] = await httpGet(
+          session,
+          `warehouse/table_columns/${nodeData.schema}/${nodeData.input_name}`
+        );
+        setSrcColumns(data.map((col: ColumnData) => col.name));
+      } catch (error) {
+        console.log(error);
+      }
+    }
+
+    if (node?.type === OPERATION_NODE) {
+      setSrcColumns(nodeData.output_cols);
+    }
+  };
+
+  const handleAddColumn = (column: string) => {
+    setSelectedColumns((prevColumns) => [...prevColumns, column]);
+  };
+
+  const handleRemoveColumn = (column: string) => {
+    setSelectedColumns((prevColumns) =>
+      prevColumns.filter((col) => col !== column)
+    );
+  };
+
+  const handleSave = async () => {
+    try {
+      const postData = {
+        op_type: operation.slug,
+        source_columns: srcColumns,
+        other_inputs: [],
+        config: { columns: {} as { [key: string]: string } },
+        input_uuid: node?.type === SRC_MODEL_NODE ? node?.data.id : '',
+        target_model_uuid: nodeData.target_model_id || '',
+      };
+
+      srcColumns.forEach((col: string) => {
+        if (selectedColumns.includes(col)) {
+          postData.config.columns[col] = col;
```

Comment:
you are sending the dictionary `{ col_name: col_name, ... }` ? shouldn't you be sending the list `[col_name, ...]`?

i think it's working because dbt-automation uses this parameter like this: `if col in columns` (`droprenamecolumns.py:25`) ... but you got lucky

Reply:
@Ishankoradia also on each new row the input box has the value from the previous row, do you know how to fix this or if not please open an issue for shamoon to do next week thank you

Reply:
Yes there should be a way. I did something similar in coalesce. Let me take a look.

Reply:
@Abhishek-N  @fatchat  i think we should be using react hook form and the `useFieldArray` hook which exposes `append` and `remove` methods out of the box to handle this. Similar logic will be used in many operations. Would be nice to get hands dirty on this library.

Also we have a custom input UI component that we have been using everywhere in the app `components/UI/Input/Input.tsx` that can be used with react hook form.  

I dont think we can clear the input without creating a new state & making autocomplete controlled which is too much work just to clear things; 

Also its okay to use mutliple autocompletes for such forms; in a loop. Will make our life easier.